test/MockSmartStore.js

```diff
@@ -314,7 +314,13 @@ var MockSmartStore = (function(window) {
                     name: "Count of soup items", 
                     example: "SELECT count(*) FROM {soupName}", 
                     pattern: /SELECT count\(\*\) FROM {(.*)}/i, 
-                    processor: this._smartQuerySoupCount 
+                    processor: this._smartQuerySoupCount
+                },
+                { 
+                    name: "Comparing soup item to integer with optional order by", 
```

Comment:
That work was mostly done in https://github.com/forcedotcom/SalesforceMobileSDK-Shared/issues/238